sklearnex/covariance/tests/test_incremental_covariance.py

```diff
@@ -163,6 +163,39 @@ def test_sklearnex_fit_on_random_data(
     assert_allclose(expected_means, result.location_, atol=1e-6)
 
 
+@pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
+@pytest.mark.parametrize("dtype", [np.float32])
+def test_pickle(dataframe, queue, dtype):
+    import pickle
+
+    from sklearnex.covariance import IncrementalEmpiricalCovariance
+
+    inccov = IncrementalEmpiricalCovariance()
+    dump = pickle.dumps(inccov)
+    inccov_loaded = pickle.loads(dump)
```

Comment:
I see with our existing pickle tests we are only pickling after fitting, is there a reason to validate pickling on just the object before fitting? is this something that should be extended to other algos?

Reply:
Just in case and to check we don't get fail after such call. for example it's theoretically possible that there might be errors in __getstate__ and __setstate__ because not all attributes are initialized

Reply:
initially this pickling didn't work, so it allowed me to fix a bug here. might make sense to check if other estimators works in this case, especially such that have C++ attributes (not sure if we have examples other than incremental algos).